### ISSUE TITLE
Replace moment with e.g. dayjs

### ISSUE THREAD

**Issue by olemp** [Puzzlepart/did#629]
- [x] I have searched for existing issues to avoid creating duplicates

### TLDR
Replace `moment` with e.g. `dayjs`.

### Description
See [Project Status](https://momentjs.com/docs/).

> Moment.js has been successfully used in millions of projects, and we are happy to have contributed to making date and time better on the web. As of September 2020, Moment gets over 12 million downloads per week! However, Moment was built for the previous era of the JavaScript ecosystem. The modern web looks much different these days. Moment has evolved somewhat over the years, but it has essentially the same design as it did when it was created in 2011. Given how many projects depend on it, we choose to prioritize stability over new features.

* [_You Probably Don't Need Moment.js Anymore_](https://dockyard.com/blog/2020/02/14/you-probably-don-t-need-moment-js-anymore) 
* [_You don't (may not) need Moment.js_](https://github.com/you-dont-need/You-Dont-Need-Momentjs/blob/master/README.md) 
* [_Why you shouldn't use Moment.js..._](https://inventi.studio/en/blog/why-you-shouldnt-use-moment-js)
* [_4 alternatives to moment.js for internationalizing dates_](https://blog.logrocket.com/4-alternatives-to-moment-js-for-internationalizing-dates/)

@damsleth suggests we look into [tc39/proposal-temporal](https://github.com/tc39/proposal-temporal)

### Current moment usage
```
129 results - 11 files

package.json:
  83      "loglevel-plugin-prefix": "0.8.4",
  84:     "moment": "2.24.0",
  85:     "moment-locales-webpack-plugin": "1.2.0",
  86      "morgan": "1.10.0",

webpack.config.js:
   5  const pkg = require('./package.json')
   6: const MomentLocalesPlugin = require('moment-locales-webpack-plugin')
   7  const CompressionPlugin = require('compression-webpack-plugin')

  83    plugins: [
  84:     new MomentLocalesPlugin({ localesToKeep: ['en-gb', 'nb'] }),
  85      new HtmlWebpackPlugin({

client/pages/Admin/Admin.tsx:
   7  import { useHistory, useParams } from 'react-router-dom'
   8: import { moment } from 'utils/date'
   9  import styles from './Admin.module.scss'

  43                      <SummaryView
  44:                         defaultSelectedYear={moment().year()}
  45                          defaultSelectedScope='weekNumber'

client/pages/Admin/SummaryView/reducer.ts:
   1  import { IContextualMenuItem } from 'office-ui-fabric-react/lib/ContextualMenu'
   2: import dateUtils, { moment } from 'utils/date'
   3  import { ISummaryViewScope, ISummaryViewState } from './types'

  47    }
  48:   if (newState.year !== moment().year()) newState.endMonthIndex = 12
  49    else newState.endMonthIndex = dateUtils.getMonthIndex()

client/pages/Admin/SummaryView/utils.tsx:
    9  import { first, unique } from 'underscore'
   10: import { moment } from 'utils/date'
   11  import { generateColumn as col } from 'utils/generateColumn'

  117      for (let i = range; i >= 0; i--) {
  118:         const key = (moment().year() - i).toString()
  119          periods.push({

client/pages/Timesheet/TimesheetPeriod.ts:
    6  import { capitalize, isBlank } from 'underscore.string'
    7: import dateUtils, { moment } from 'utils/date'
    8  import { ITimesheetParams } from './types'

   61    /**
   62:    * Start date time moment object
   63     */
   64:   private _startDateTime?: moment.Moment
   65  
   66    /**
   67:    * End date time moment object
   68     */
   69:   private _endDateTime?: moment.Moment
   70  

  101      this._month = capitalize(_period.month)
  102:     this._startDateTime = moment(_period.startDateTime)
  103:     this._endDateTime = moment(_period.endDateTime)
  104      this.isConfirmed = _period.isConfirmed

client/pages/Timesheet/TimesheetScope.ts:
   1: import dateUtils, { moment } from 'utils/date'
   2  import { ITimesheetParams } from './types'

   4  export interface ITimesheetScopeOptions {
   5:   amount: moment.DurationInputArg1
   6:   unit: moment.unitOfTime.DurationConstructor
   7  }

  14  export class TimesheetScope {
  15:   private _startDateTime?: moment.Moment
  16:   private _endDateTime?: moment.Moment
  17  

  23    constructor(value?: ITimesheetParams | string) {
  24:     let start = moment()
  25      if (value) {

  27          const startIsValid = !isNaN(Date.parse(value))
  28:         if (startIsValid) start = moment(value)
  29        } else {
  30:         start = moment().year(parseInt(value.year)).week(parseInt(value.week)).startOf('isoWeek')
  31        }

  58     */
  59:   private _set(start: moment.Moment) {
  60      this._startDateTime = dateUtils.startOfWeek(start)

  81     */
  82:   public getDay(index: number): moment.Moment {
  83:     return this._startDateTime.clone().add(index, 'days' as moment.DurationInputArg2)
  84    }

  86    public get isCurrentWeek(): boolean {
  87:     return this._startDateTime.week() === moment().week()
  88    }

client/utils/date.ts:
    1  import { TFunction } from 'i18next'
    2: import moment from 'moment'
    3  import { format } from 'office-ui-fabric-react/lib/Utilities'

    6  
    7: export { moment }
    8  
    9  export default new (class DateUtils {
   10:   private _momentLocale: string
   11  

   17    public setup(locale: string) {
   18:     this._momentLocale = locale
   19:     moment.locale(this._momentLocale)
   20    }

   22    /**
   23:    * Converts date string to moment, adding timezone offset
   24     *

   26     */
   27:   toMoment(date: string) {
   28:     const m = moment(date)
   29      return m.add(m.toDate().getTimezoneOffset(), 'minutes')

   56    formatDate(date: string, dateFormat: string): string {
   57:     const m = moment.utc(date)
   58      return m.add(-m.toDate().getTimezoneOffset(), 'minutes').format(dateFormat)

   63     *
   64:    * @param {string | Date | moment.Moment} date Date string
   65     */
   66:   startOfWeek(date?: string | Date | moment.Moment): moment.Moment {
   67:     const m = moment.utc(date)
   68      return m.add(-m.toDate().getTimezoneOffset(), 'minutes').startOf('isoWeek')

   75     */
   76:   endOfWeek(date?: string | Date | moment.Moment): moment.Moment {
   77:     const m = moment.utc(date)
   78      return m.add(-m.toDate().getTimezoneOffset(), 'minutes').endOf('isoWeek')

   83     *
   84:    * @param {moment.Moment} start Start
   85:    * @param {moment.Moment} end End
   86     * @param {string} dayFormat Date format
   87     */
   88:   getDays(start: moment.Moment, end: moment.Moment, dayFormat: string): string[] {
   89      const days = []
   90      for (let i = 0; i <= end.weekday() - start.weekday(); i++) {
   91:       days.push(capitalize(start.clone().add(i, 'days').locale(this._momentLocale).format(dayFormat)))
   92      }

  101    public addMonth(amount = 1) {
  102:     return moment().add(amount, 'month')
  103    }

  110    public subtractMonths(amount = 1) {
  111:     return moment().subtract(amount, 'month')
  112    }

  123     */
  124:   public getMonthYear(date: moment.Moment = moment()) {
  125      return {

  146    getMonthName(monthIndex?: number, format = 'MMMM', captialize = false): string {
  147:     const date = moment().locale(this._momentLocale)
  148      let name: string

  157     *
  158:    * @param {moment.Moment} start Start
  159:    * @param {moment.Moment} end End
  160     * @param {object} options Options

  162    getTimespanString(
  163:     start: moment.Moment,
  164:     end: moment.Moment,
  165      options: Record<string, any> = {

  172      return start
  173:       .locale(this._momentLocale)
  174:       ['twix'](end.locale(this._momentLocale), { allDay: true })
  175        .format(options)

  183      return Array.apply(0, Array(12)).map((_: any, i: number) => {
  184:       return capitalize(moment().month(i).format('MMMM'))
  185      })

  188    /**
  189:    * Get a string representation of the moment date instance
  190     *
  191:    * @param {moment.Moment} date Moment date
  192     *

  194     */
  195:   toString(date: moment.Moment): string {
  196      return date.toISOString().replace('Z', '')

  202    getWeek(): number {
  203:     return moment().week()
  204    }

  215    getMonthIndex(): number {
  216:     return moment().month() + 1
  217    }

  222    getYear(): number {
  223:     return moment().year()
  224    }

client/utils/exportExcel.ts:
   4  import { loadScripts } from './loadScripts'
   5: import { moment } from 'utils/date'
   6  

  52                  return {
  53:                   v: moment(fieldValue).format('YYYY-MM-DD HH:mm'),
  54                    t: 'd'

server/api/graphql/resolvers/timesheet.utils.ts:
   4  import get from 'get-value'
   5: import moment from 'moment'
   6  

   9   *
  10:  * @param {string | moment.Moment} startDateTime Start date time in ISO format
  11:  * @param {string | moment.Moment} endDateTime End date time in ISO format
  12:  * @param {string} locale User locale for moment formatting
  13   */
  14  export function getPeriods(
  15:   startDateTime: string | moment.Moment,
  16:   endDateTime: string | moment.Moment,
  17    locale: string

server/utils/index.ts:
    1: import moment from 'moment'
    2  import stripHtml from 'string-strip-html'

   17  export const getDurationHours = (startDateTime: string, endDateTime: string): number => {
   18:   return moment.duration(moment(endDateTime).diff(moment(startDateTime))).asHours()
   19  }

   26  export const getPeriod = (date: any) => {
   27:   const d = moment(date)
   28    return [d.isoWeek(), d.month() + 1, d.year()].join('_')

   33   *
   34:  * @param {string | moment.Moment} date Date
   35   */
   36: export const getWeek = (date?: string | moment.Moment) => {
   37:   return moment(date).isoWeek()
   38  }

   45  export const getYear = (date?: string) => {
   46:   return moment(date).year()
   47  }

   51   *
   52:  * NOTE: Need to add +1 since moment.month is zero-indexed
   53   *

   56  export const getMonthIndex = (date: any) => {
   57:   return moment(date).month() + 1
   58  }

   65  export const startOfMonth = (date: any) => {
   66:   const d = moment(date).startOf('month')
   67    return d.toISOString().replace('Z', '')

   75  export const endOfMonth = (date: any) => {
   76:   const d = moment(date).endOf('month')
   77    return d.toISOString().replace('Z', '')

   85  export const startOfWeek = (week: any) => {
   86:   return moment().week(week).startOf('isoWeek')
   87  }

   94  export const endOfWeek = (week: any) => {
   95:   return moment().week(week).endOf('isoWeek')
   96  }

  105  export const formatDate = (date: any, dateFormat: any, locale: any, timeZone = 'Europe/Oslo') => {
  106:   return (moment(date) as any).locale(locale).tz(timeZone).format(dateFormat)
  107  }

  114  export const isAfterToday = (date: any) => {
  115:   return moment(date).isAfter(moment())
  116  }

```
